controller/customdomain_controller_test.go

```diff
@@ -62,6 +69,82 @@ func TestCustomDomainController(t *testing.T) {
 		},
 	}
 
+	// A CustomDomain resource with routeSelector nil.
+	customdomainRouteSelectorNil := &customdomainv1alpha1.CustomDomain{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      instanceNameRouteSelectorNil,
+			Namespace: userNamespace,
+		},
+		Spec: customdomainv1alpha1.CustomDomainSpec{
+			Domain: userDomain,
+			Scope:  instanceScope,
+			Certificate: corev1.SecretReference{
+				Name:      userSecretName,
+				Namespace: userNamespace,
+			},
+			RouteSelector: &v1.LabelSelector{
```

Comment:
I think you want to set RouteSelector to nil
```suggestion
			RouteSelector: nil
```